app/src/main/java/fr/free/nrw/commons/nearby/RotationGestureOverlay.java

```diff
@@ -0,0 +1,129 @@
+package fr.free.nrw.commons.nearby;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import org.osmdroid.views.MapView;
+import org.osmdroid.views.overlay.IOverlayMenuProvider;
+import org.osmdroid.views.overlay.Overlay;
+import org.osmdroid.views.overlay.gestures.RotationGestureDetector;
+
+public class RotationGestureOverlay extends Overlay implements
+    RotationGestureDetector.RotationListener, IOverlayMenuProvider {
+    private final static boolean SHOW_ROTATE_MENU_ITEMS = false;
+
+    private final static int MENU_ENABLED = getSafeMenuId();
+    private final static int MENU_ROTATE_CCW = getSafeMenuId();
+    private final static int MENU_ROTATE_CW = getSafeMenuId();
```

Comment:
Sorry to not have conveyed this earlier: Map rotation is not really needed. Feel free to not remove it, especially if rotation makes the code more complex. Tiles seem not often not update after rotation, which impacts readability, so no rotation might be better.

Reply:
Oh no worries, I have removed rotation feature in newer commit